info-compiler/src/main/java/com/google/sps/infocompiler/InfoCompiler.java

```diff
@@ -270,28 +274,39 @@ void storeElectionContestInDatabase(String electionQueryId, JsonObject contest)
     datastore.update(electionEntity);
   }
 
-  private boolean checkIncumbency(String candidateName, String position, String division) {
+  private boolean checkIncumbency(Map<String, String> map, String position, String name) {
+    for (Map.Entry<String, String> e : map.entrySet()) {
+      if (e.getKey() == position && e.getValue() == name) return true;
+    }
+    return false;
+  }
+  private Map<String, String> getincumbents(String division) {
+    Map<String, String> mapOfIncumbents = new HashMap<>();  
     try {
       String queryUrl = String.format("%s&ocdDivisionId=%s", REPRENTATIVE_QUERY_URL, division);
       JsonObject representatives = queryCivicInformation(queryUrl);
       JsonArray offices = representatives.getAsJsonArray("offices");
-      int i = 0;
-      for (JsonElement eachOffice : offices){
-        JsonObject office = (JsonObject) eachOffice;
-        String officeName = office.get("name").getAsString();
-        if (position.equals(officeName)){
-          break;
+      JsonObject office;
+      String officeName;
+      JsonArray officialIndicesArray;
+      JsonArray officials;
+      for (JsonElement eachOffice : offices) {
+        office = (JsonObject) eachOffice;
+        officeName = office.get("name").getAsString();
+        officialIndicesArray =  office.getAsJsonArray("officialIndices");
+        officials = office.getAsJsonArray("officials");
+        for (int i = 0; i < officialIndicesArray.size(); i++) {
+          int j = officialIndicesArray.get(i).getAsInt();
+          mapOfIncumbents.put(officeName,officials.get(i).getAsString());
```

Comment:
nit: space after comma

Reply:
Can there be more than one official per office?  If so, then this will overwrite any official except the last.  Use a `Map<String, List<String>>` instead if there can be multiple.